airflow/utils/operator_helpers.py

```diff
@@ -276,10 +277,14 @@ def _run():
 
             for metadata in _run():
                 if isinstance(metadata, Metadata):
-                    outlet_events[metadata.uri].extra.update(metadata.extra)
+                    # TODO: handle asset name
+                    outlet_events[Asset(uri=metadata.uri)].extra.update(metadata.extra)
 
                     if metadata.alias_name:
-                        outlet_events[metadata.alias_name].add(metadata.uri, extra=metadata.extra)
+                        # TODO: handle asset name
+                        outlet_events[AssetAlias(name=metadata.alias_name)].add(
+                            Asset(uri=metadata.uri), extra=metadata.extra
+                        )
```

Comment:
Let’s deprecate passing in a URI to Metadata in 2.11 and require an Asset there as well.

Reply:
remove string access and create a deprecation PR https://github.com/apache/airflow/pull/44791